frontend/src/components/admin/assessment-status/EditStatusButtons/PreviewButton.tsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { useHistory } from "react-router-dom";
+import { useQuery } from "@apollo/client";
+
+import { GET_TEST } from "../../../../APIClients/queries/TestQueries";
+import type { TestResponse } from "../../../../APIClients/types/TestClientTypes";
+import * as Routes from "../../../../constants/Routes";
+import useToast from "../../../common/info/useToast";
+import PopoverButton from "../../../common/popover/PopoverButton";
+
+interface PreviewButtonProps {
+  assessmentId: string;
+}
+
+const PreviewButton = ({
+  assessmentId,
+}: PreviewButtonProps): React.ReactElement => {
+  const history = useHistory();
+  const { data } = useQuery<{ test: TestResponse }>(GET_TEST, {
```

Comment:
Iirc the tests table already loads this data for each row. Is there any way you could reuse that instead? Otherwise each user will be hammering the backend twice for each test that exists in the database

Reply:
Hmmm, but does it? Because the table runs the `GET_ALL_TESTS` query, right? Which is all the test metadata without the specific questions.

Here, we call `GET_TEST` which queries for the test metadata + questions for one _specific_ test. I think this is similar to the implementation of `EditButton`.

Let me know if I am misunderstanding your comment here!

Reply:
Right. I'm wondering if it would be better to just add the additional data to the GET_ALL_TESTS query? Since imho our concern here might not be bandwidth but compute scale, since we have a single-replica base tier backend

Reply:
I guess I'm also thinking that EditButton could use a different implementation